inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,249 @@
+        source='crossref'
+    )
+
+    builder.add_if_non_empty(
+        'license',
+        url=form_fields.get('license_url')
+    )
+
+    builder.add_if_non_empty(
+        'public_note',
+        public_notes=form_fields.get('public_notes')
+    )
+
+    builder.add_if_non_empty(
+        'public_note',
+        public_notes=form_fields.get('note'),
+        source='arXiv' if form_fields.get('categories') else 'CrossRef'
+    )
+
+    note = 'Presented on {0}'.format(form_fields.get('defense_date')) if \
+        form_fields.get('defense_date') else None
+    builder.add_if_non_empty(
+        'public_note',
+        public_notes=note
+    )
+
+    builder.add_if_non_empty(
+        'url',
+        url=form_fields.get('url')
+    )
+
+    builder.add_if_non_empty(
+        'url',
+        url=form_fields.get('additional_url')
+    )
+
+    [builder.add_if_non_empty(
+        'report_number',
+        report_number=report_number.get('report_number')
+    ) for report_number in form_fields.get('report_numbers', [])]
+
+    builder.add_if_non_empty(
+        'collaboration',
+        collaboration=form_fields.get('collaboration')
+    )
+
     try:
-        email = User.query.get(userid).email
+        email = User.query.get(obj.id_user).email
     except AttributeError:
-        email = ''
-    try:
-        # TODO Make sure we are getting the ORCID id correctly
-        source = UserIdentity.query.filter_by(id_user=userid, method='orcid').one()
-    except NoResultFound:
-        source = ''
-    if source:
-        source = source.method + ':' + source.id
-    data['acquisition_source'] = dict(
-        source=source,
+        email = None
+
+    orcid = normalized_orcid(obj.id_user)
+
+    builder.add_if_non_empty(
+        'acquisition_source',
+        submission_number=obj.id,
         email=email,
-        date=date.today().isoformat(),
-        method="submission",
-        submission_number=str(obj.id),
+        orcid=orcid,
+        method='submitter'
     )
-    # ==============
-    # Extra comments
-    # ==============
-    if form_fields.get('extra_comments'):
-        data.setdefault('hidden_notes', []).append(
-            {
-                'value': form_fields['extra_comments'],
-                'source': 'submitter'
-            }
-        )
-    # ==========================
-    # Journal name normalization
-    # ==========================
-    journal_title = get_value(data, 'publication_info[0].journal_title')
-    if journal_title:
-        hits = JournalsSearch().query(
-            'match', title_variants__title__lowercased=journal_title).execute()
-
-        if hits:
-            try:
-                short_title = hits[0].short_titles[0].title
-                data['publication_info'][0]['journal_title'] = short_title
-            except (AttributeError, IndexError):
-                pass
-
-    # Finally, return the converted data
-    return data
+
+    builder.validate_object()
```

Comment:
Shall we handle this gracefully? Otherwise the user will get a 500 error. If this call raises an exception, you can catch it in `literaturesuggest/views.py` and show to the user an informative error message.